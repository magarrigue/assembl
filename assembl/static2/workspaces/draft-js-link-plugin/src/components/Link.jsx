// @flow
import * as React from 'react';

type Props = {
  children: React.Node,
  className: string,
  entityKey: string,
  getEditorState: Function
};

const Link = ({ children, className, entityKey, getEditorState }: Props) => {
  const entity = getEditorState()
    .getCurrentContent()
    .getEntity(entityKey);
  const entityData = entity ? entity.get('data') : undefined;
  const href = (entityData && entityData.url) || undefined;
  const target = (entityData && entityData.target) || undefined;
  const title = (entityData && entityData.title) || undefined;
  return (
    <a className={className} title={title} href={href} target={target} rel="noopener noreferrer">
      {children}
    </a>
  );
};

export default Link;