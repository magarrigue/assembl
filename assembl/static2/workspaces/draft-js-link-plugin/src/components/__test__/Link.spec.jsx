import React from 'react';
import renderer from 'react-test-renderer';
import { ContentState, EditorState, Modifier, SelectionState } from 'draft-js';

import Link from '../Link';

describe('Link component', () => {
  it('should render a link', () => {
    // create an editor state with a link entity
    let contentState = ContentState.createFromText('I will input the solid state PNG port');
    contentState = contentState.createEntity('LINK', 'MUTABLE', {
      target: '_blank',
      title: 'My link',
      url: 'https://en.wikipedia.org/wiki/Portable_Network_Graphics'
    });
    const contentBlock = contentState.getFirstBlock();
    const blockKey = contentBlock.getKey();
    const selection = SelectionState.createEmpty(blockKey).merge({
      anchorKey: blockKey,
      anchorOffset: '30',
      focusKey: blockKey,
      focusOffset: '32'
    });
    const entityKey = contentState.getLastCreatedEntityKey();
    contentState = Modifier.applyEntity(contentState, selection, entityKey);
    let editorState = EditorState.createWithContent(contentState);
    editorState = EditorState.acceptSelection(editorState, selection);

    const getEditorStateMock = () => editorState;
    const props = {
      children: 'PNG',
      className: 'link',
      entityKey: entityKey,
      getEditorState: getEditorStateMock
    };
    const component = renderer.create(<Link {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});