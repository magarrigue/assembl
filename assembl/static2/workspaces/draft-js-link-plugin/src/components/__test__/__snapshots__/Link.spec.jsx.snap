// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component should render a link 1`] = `
<a
  className="link"
  href="https://en.wikipedia.org/wiki/Portable_Network_Graphics"
  rel="noopener noreferrer"
  target="_blank"
  title="My link"
>
  PNG
</a>
`;
